### PR TITLE
interfaces/x11: add slot side apparmor rules on app slots too

### PR DESCRIPTION
**Pull request by sd-hd**
On classic systems these apparmor rules were not loaded.

To create a x11-server in a strict snap, these rules are needed.

### CODE REVIEW

File: interfaces/builtin/x11.go [hunk 1]
```diff
@@ -171,9 +169,7 @@ func (iface *x11Interface) SecCompPermanentSlot(spec *seccomp.Specification, slo
 }
 
 func (iface *x11Interface) AppArmorPermanentSlot(spec *apparmor.Specification, slot *snap.SlotInfo) error {
-	if !release.OnClassic {
-		spec.AddSnippet(x11PermanentSlotAppArmor)
-	}
+	spec.AddSnippet(x11PermanentSlotAppArmor)
```
Comment (jdstrand) [inline interfaces/builtin/x11.go]: Please use:
```
        // Only apply slot snippet when running as application snap
        // on classic, slot side can be system or application
        if !implicitSystemPermanentSlot(slot) {
                spec.AddSnippet(x11PermanentSlotAppArmor)
        }
```

File: interfaces/builtin/x11.go [hunk 2]
```diff
@@ -154,12 +154,10 @@ type x11Interface struct {
...
-	if !release.OnClassic {
-		old := "###PLUG_SECURITY_TAGS###"
-		new := plugAppLabelExpr(plug)
-		snippet := strings.Replace(x11ConnectedSlotAppArmor, old, new, -1)
-		spec.AddSnippet(snippet)
-	}
+	old := "###PLUG_SECURITY_TAGS###"
+	new := plugAppLabelExpr(plug)
+	snippet := strings.Replace(x11ConnectedSlotAppArmor, old, new, -1)
+	spec.AddSnippet(snippet)
```
Comment (jdstrand) [inline interfaces/builtin/x11.go]: Please use:
```
        // Only apply slot snippet when running as application snap
        // on classic, slot side can be system or application
        if !implicitSystemConnectedSlot(slot) {
                old := "###PLUG_SECURITY_TAGS###"
                new := plugAppLabelExpr(plug)
                snippet := strings.Replace(x11ConnectedSlotAppArmor, old, new, -1)
                spec.AddSnippet(snippet)
        }
        return nil
```

File: interfaces/builtin/x11_test.go [hunk 3]
```diff
@@ -113,28 +114,6 @@ func (s *X11InterfaceSuite) TestAppArmorSpec(c *C) {
...
-	spec = &apparmor.Specification{}
-	c.Assert(spec.AddConnectedSlot(s.iface, s.plug, s.classicSlot), IsNil)
-	c.Assert(spec.SecurityTags(), HasLen, 0)
-
-	// permanent classic slot
-	spec = &apparmor.Specification{}
-	c.Assert(spec.AddPermanentSlot(s.iface, s.classicSlotInfo), IsNil)
-	c.Assert(spec.SecurityTags(), HasLen, 0)
-}
-
```
Comment (jdstrand) [inline interfaces/builtin/x11_test.go]: Please adjust to add new tests. You can see how that was done in avahi_observe_test.go.

**Comment by sd-hd** [comment]
@jdstrand 
thank you :-)
maybe tomorrow i can have a look ;-)

File: interfaces/builtin/x11.go (hunk 1 above)
Comment (sd-hd) [reply interfaces/builtin/x11.go]: done, on next push --force

File: interfaces/builtin/x11_test.go (hunk 3 above)
Comment (sd-hd) [reply interfaces/builtin/x11_test.go]: today, at the moment. i didn't understand. maybe tomorrow ;-)

File: interfaces/builtin/x11.go (hunk 2 above)
Comment (sd-hd) [reply interfaces/builtin/x11.go]: done

File: interfaces/builtin/x11_test.go (hunk 3 above)
Comment (sd-hd) [reply interfaces/builtin/x11_test.go]: ok. today, i think i have fixed. Not sure.

**Comment by sd-hd** [comment]
> Thanks for the changes. This is close, please see inline for what I meant about the new test case.

Thank you

Hope my latest changes are correct now ;-)

File: interfaces/builtin/x11_test.go [hunk 4]
```diff
@@ -122,17 +123,20 @@ func (s *X11InterfaceSuite) TestAppArmorSpecOnClassic(c *C) {
...
+	c.Assert(spec.SecurityTags(), DeepEquals, []string{"snap.x11.app1"})
+	c.Assert(spec.SnippetForTag("snap.x11.app1"), testutil.Contains, `peer=(label="snap.consumer.app"),`)
 
 	// permanent classic slot
 	spec = &apparmor.Specification{}
-	c.Assert(spec.AddPermanentSlot(s.iface, s.classicSlotInfo), IsNil)
-	c.Assert(spec.SecurityTags(), HasLen, 0)
+	c.Assert(spec.AddPermanentSlot(s.iface, s.coreSlotInfo), IsNil)
+	c.Assert(spec.SecurityTags(), DeepEquals, []string{"snap.x11.app1"})
+	c.Assert(spec.SnippetForTag("snap.x11.app1"), testutil.Contains, "capability sys_tty_config,")
```
[the remark this answers is not in the document]
Comment (sd-hd) [reply interfaces/builtin/x11_test.go]: done

**Comment by zyga** [comment]
@sd-hd would you mind merging master _or_ rebasing on master please?

**Comment by sd-hd** [comment]
@zyga hey, yes! I help ;-) ever...

**Comment by sd-hd** [comment]
@zyga:
Why does this happen in the the Unit-Test?

```
FAIL: cmd_sign_build_test.go:108: SnapSignBuildSuite.TestSignBuildWorksDevelGrade

cmd_sign_build_test.go:126:
    c.Assert(err, IsNil)
... value *errors.errorString = &errors.errorString{s:"cannot sign assertion: cannot sign assertion: bad GPG produced signature: it does not verify: openpgp: invalid signature: RSA verification failure"} ("cannot sign assertion: cannot sign assertion: bad GPG produced signature: it does not verify: openpgp: invalid signature: RSA verification failure")
```

**Comment by anonymouse64** [comment]
> Why does this happen in the the Unit-Test?
> ```
> FAIL: cmd_sign_build_test.go:108: SnapSignBuildSuite.TestSignBuildWorksDevelGrade
> 
> cmd_sign_build_test.go:126:
>     c.Assert(err, IsNil)
> ... value *errors.errorString = &errors.errorString{s:"cannot sign assertion: cannot sign assertion: bad GPG produced signature: it does not verify: openpgp: invalid signature: RSA verification failure"} ("cannot sign assertion: cannot sign assertion: bad GPG produced signature: it does not verify: openpgp: invalid signature: RSA verification failure")

this is a transient failure we have not had time to debug fully. If you rebase this on master, we have made a number of changs to GitHub actions since this was opened, so rebasing on master would probably make the unit test failure go away (and if it doesn't we're happy to restart tests for you).

**Comment by anonymouse64** [comment]
This change seems to have been picked up and merged to master with #9530, so I am going to close this PR. If there are further changes you would like to see made, please don't hesitate to comment here or open a new PR. Thanks

**Pull request closed without merging**